zcash_client_sqlite/src/wallet/db.rs

```diff
@@ -236,12 +318,20 @@ CREATE TABLE sapling_tree_cap (
     cap_data BLOB NOT NULL
 )";
 
+/// Stores the checkpointed positions in the Sapling [`ShardTree`].
+///
+/// TODO: Document this.
+///
+/// [`ShardTree`]: shardtree::ShardTree
 pub(super) const TABLE_SAPLING_TREE_CHECKPOINTS: &str = "
 CREATE TABLE sapling_tree_checkpoints (
     checkpoint_id INTEGER PRIMARY KEY,
     position INTEGER
 )";
 
+/// TODO: Summary of what this stores.
+///
+/// TODO: Document this.
```

Comment:
```suggestion
/// Stores metadata about the positions of Sapling notes that have been spent but for which
/// witness information as not yet been removed from the note commitment tree.
///
/// In the process of updating the note commitment tree in response to the addition of 
/// a block, it is necessary to temporarily continue to store witness information for
/// each note so that a spent note can be made spendable again after a rollback of the
/// spending block. This table caches the metadata needed for that restoration.
```